src/definitions.jl

```diff
@@ -396,6 +396,49 @@ function ifftshift(x,dim)
     circshift(x, s)
 end
 
+##############################################################################
+
+
+struct Frequencies{T<:Number} <: AbstractVector{T}
+    nreal::Int
+    n::Int
+    multiplier::T
+end
+
+unsafe_getindex(x::Frequencies, i::Int) =
+    (i-1+ifelse(i <= x.nreal, 0, -x.n))*x.multiplier
```

Comment:
This seems like it should be `i-1 ≤ x.nreal`

Reply:
Oh, I guess by `nreal` you mean the number of nonnegative frequencies, in which case `i ≤ x.nreal` is right.   But maybe call it `n_nonnegative` or something like that to be clearer?